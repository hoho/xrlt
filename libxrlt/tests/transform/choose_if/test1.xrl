<?xml version="1.0"?>
<xrl:requestsheet xmlns:xrl="http://xrlt.net/Transform">

    <xrl:response>
        <xrl:variable name="tmp">
            <xrl:include>
                <xrl:href>/tmp</xrl:href>
                <xrl:type>text</xrl:type>
            </xrl:include>
        </xrl:variable>

        <xrl:choose>
            <xrl:when test="false()">aa</xrl:when>
            <xrl:when test="false()">bb</xrl:when>
            <xrl:when test="true()">cc</xrl:when>
            <xrl:when test="true()">dd</xrl:when>
            <xrl:otherwise>ee</xrl:otherwise>
        </xrl:choose>

        <xrl:if test="false()">ff</xrl:if>

        <xrl:if test="true()">gg</xrl:if>

        <xrl:choose>
            <xrl:when test="false()">hh</xrl:when>
            <xrl:when test="false()">ii</xrl:when>
            <xrl:otherwise>jj</xrl:otherwise>
        </xrl:choose>

        <xrl:choose>
            <xrl:when test="false()">kk</xrl:when>
            <xrl:when test="$tmp = 'ololo'">ll</xrl:when>
            <xrl:when test="$tmp = 'alala'">mm</xrl:when>
        </xrl:choose>

        <xrl:if test="$tmp = 'ololo'">nn</xrl:if>

        <xrl:if test="$tmp = 'alala'">oo</xrl:if>
    </xrl:response>

</xrl:requestsheet>
