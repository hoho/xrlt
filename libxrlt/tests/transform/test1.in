id:0, type:0, last:0, error: 0, data:
id:0, type:0, last:0, error: 0, data:
id:0, type:0, last:0, error: 0, data:
id:0, type:0, last:0, error: 0, data:
id:3, type:2, last:0, error: 0, data:{"aaa
id:3, type:2, last:1, error: 0, data:bbb": 123, "ccc": true}
id:0, type:0, last:0, error: 0, data:
id:2, type:2, last:0, error: 0, data:<hello>
id:2, type:2, last:0, error: 0, data:world
id:2, type:2, last:1, error: 0, data:</hello>
id:0, type:0, last:0, error: 0, data:
id:1, type:2, last:0, error: 0, data:there are
id:1, type:2, last:0, error: 0, data: many clouds
id:1, type:2, last:1, error: 0, data: in the sky
id:0, type:0, last:0, error: 0, data:
