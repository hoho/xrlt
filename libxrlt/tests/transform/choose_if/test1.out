XRLT_STATUS_SUBREQUEST
sr id: 1
sr method: GET
sr type: TEXT
sr url: /tmp
sr query: (null)
sr body: (null)
XRLT_STATUS_CHUNK
chunk: cc
chunk: gg
chunk: jj
XRLT_STATUS_WAITING
XRLT_STATUS_CHUNK
chunk: mm
chunk: oo
XRLT_STATUS_DONE
